purekit/src/main/kotlin/com/virgilsecurity/purekit/pure/PureCrypto.java

```diff
@@ -73,10 +98,12 @@ PureCryptoData encryptCellKey(byte[] plainTextData,
 
             cipher.addSigner(signingKey.getIdentifier(), signingKey.getPrivateKey());
 
+            //TODO add null check for recipients
             for (VirgilPublicKey key : recipients) {
                 cipher.addKeyRecipient(key.getIdentifier(), key.getPublicKey());
             }
 
+            //TODO Sha512 is not closed, but it's OK because native resources will be cleared on finalize
```

Comment:
```
try (Sha512 sha512 = new Sha512()) {
    cipher.setSignerHash(sha512);
}
```
Would be better?

Reply:
Just create it in the same try() as RecipientCipher